app/queries/district_analytics_query.rb

```diff
@@ -73,6 +91,46 @@ def patients_with_bp_by_period
     group_by_facility_and_date(@patients_with_bp_by_period, :patients_with_bp_by_period)
   end
 
+  def follow_up_patients_by_period
```

Comment:
This query is doing some crazy things, I think we should re-think this.

It tries hard to assign a single visit to any user to match numbers in the facility reports. If a patient visits 2 separate users in the same month and the same facility, it only picks one of those users (the first one) and ends up not assigning the other user a visit. 

This metric is dubious at best because it under-counts the user's work.

I think we should simplify this and make it represent the truth better; multiple visits across users should be counted for all users.

Reply:
Ooh that's a good call. Counting both visits is preferable, and I didn't realize that it would be easier to do so. I think we can avoid a lot of this `from`, `distinct` convolution.

Let me take a stab at this as well.

Reply:
I might be able to dig the original original query.

Reply:
This is helpful, thanks. Yup, I see a `hypertension_followups_*` scope in our code today that seems to make this a lot easier. Stay tuned 🤞

Reply:
@kitallis @harimohanraj89 thanks for the feedback. I updated the district and facility `follow_up_patients_by_period` queries to use the first version of the query referenced above by Kitty. Rob and I also updated the respective query specs.

Reply:
END OF DAY UPDATE: @harimohanraj89 @kitallis there was a discussion on Slack with Daniel (thread starts here)[https://simpledotorg.slack.com/archives/CK95QFJA2/p1603225062040200] and we'd like to update the follow-ups query to count unique patient visits with a BP recorded per day within a month. I updated the footnotes to reflect this new query, here's what's pending:
* Update district and facility follow-up queries so that BPs are unique by day
* Create respective tests (I took a stab at creating the facility test)


Would love to pair with either of you tomorrow if you have a chance so we can ship tomorrow

Reply:
@harimohanraj89 @claudiovallejo I'm a little confused about this new definition, we can discuss this in person more clearly, but:

In either district or facility follow-ups, the idea of follow-ups has had nothing to do with visits up until now, plainly stated follow-ups are like this:

```
Have registered before the beginning of month M at any facility
AND
Have diagnoed HTN = yes
AND
Have assigned facility at facility F 
AND
Have at least 1 Blood Pressure reading in month M at facility F
```

If we do it by visits, then if a Patient visits 30 times in a month each day, the follow-ups for that month will be 30. That historically has never been how we've computed follow-ups. See the [scenarios](https://docs.google.com/document/d/16ga4Ug-t-aBvy_p0OfxbOacBDgH7LdIC7VCP4N1y_Cg/edit#heading=h.5whpx7awcvca) here for reference.

Secondly, how are we planning to group these numbers in the facility analytics, by user?

Reply:
Closing this loop after oodles of discussion. Rather than nail this definition the first time around, let's instead aim for _consistency_ across facility details, district details, and progress tab. This PR is introducing some centralized service classes to fetch this info, so we can iterate on the definition easily.